zerver/views/report.py

```diff
@@ -72,7 +79,8 @@ def json_report_error(request, user_profile, message=REQ, stacktrace=REQ,
                       ui_message=REQ(validator=check_bool), user_agent=REQ,
                       href=REQ, log=REQ,
                       more_info=REQ(validator=check_dict([]), default=None)):
-
+    # type: (Any, UserProfile, Any, Any, REQ, Any, Any, Any, REQ) -> HttpResponse
+    # TODO: INCOMPATIBLE_TYPES_WITH_REQ_TYPE_OBJECT_AND_REQ
```

Comment:
The other thing that's kinda not ideal about the way REQ is interacting with mypy for the views is that ideally, we'd be able to have mypy be checking the things we actually care about, namely the types of the arguments to the function constructed by the decorators here.

E.g. inside the body of the `json_report_error` function here, `ui_message` is a bool and ideally the type checker would know that...